DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/DateShiftProcessor.cs

```diff
@@ -0,0 +1,86 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Model;
+using Microsoft.Health.Dicom.Anonymizer.Core.Processors.Settings;
+using Microsoft.Health.Dicom.DeID.SharedLib;
+using Microsoft.Health.Dicom.DeID.SharedLib.Settings;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class DateShiftProcessor : IAnonymizerProcessor
+    {
+        private readonly DateShiftFunction _dateShiftFunction;
+        private readonly DateShiftScope _dateShiftScope = DateShiftScope.SopInstance;
+
+        public DateShiftProcessor(JObject settingObject, IAnonymizerSettingsFactory settingFactory = null)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            settingFactory ??= new AnonymizerSettingsFactory();
+            var dateShiftSetting = settingFactory.CreateAnonymizerSetting<DateShiftSetting>(settingObject);
+            _dateShiftFunction = new DateShiftFunction(dateShiftSetting);
+            if (settingObject.TryGetValue("DateShiftScope", StringComparison.OrdinalIgnoreCase, out JToken scope))
+            {
+                _dateShiftScope = (DateShiftScope)Enum.Parse(typeof(DateShiftScope), scope.ToString());
+            }
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+            EnsureArg.IsNotNull(context, nameof(context));
+
+            _dateShiftFunction.DateShiftKeyPrefix = _dateShiftScope switch
+            {
+                DateShiftScope.StudyInstance => context.StudyInstanceUID ?? string.Empty,
```

Comment:
>context.StudyInstanceUID ?? string.Empty [](http://example.com/codeflow?start=48&length=40)

In AnonymizerEngine::InitContext, all fields are guaranteed to be not null.
So the check is unnecessary unless there're other places where the context object could be created.

Reply:
The process function is public that can be used by others. We could not guarantee the context don't has null variable.